ddtrace/contrib/molten/wrappers.py

```diff
@@ -0,0 +1,91 @@
+import wrapt
+import molten
+
+from ... import Pin
+from ...utils.importlib import func_name
+
+def trace_wrapped(resource, wrapped, *args, **kwargs):
+    pin = Pin.get_from(molten)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    with pin.tracer.trace(func_name(wrapped), service=pin.service, resource=resource):
+        return wrapped(*args, **kwargs)
+
+
+def trace_func(resource):
+    """Trace calls to function using provided resource name
+    """
+    @wrapt.function_wrapper
+    def _trace_func(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+
+        if not pin or not pin.enabled():
+            return wrapped(*args, **kwargs)
+
+        with pin.tracer.trace(func_name(wrapped), service=pin.service, resource=resource):
+            return wrapped(*args, **kwargs)
+
+    return _trace_func
+
+
+class WrapperComponent(wrapt.ObjectProxy):
+    """ Tracing of components """
+    def can_handle_parameter(self, *args, **kwargs):
+        func = self.__wrapped__.can_handle_parameter
+        cname = func_name(self.__wrapped__)
+        resource = '{}.{}'.format(cname, func.__name__)
+        return trace_wrapped(resource, func, *args, **kwargs)
+
+    # TODO[tahir]: the signature of a wrapped resolve method causes DIError to
```

Comment:
oof... and this is probably the more important method to trace too, as opposed to `can_hanndle_parameter`?

Reply:
Agreed, let's put it on the backlog!